### PR TITLE
layers: Add 07791

### PR DESCRIPTION
**Pull request by arno-lunarg**
Part of https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/7361

### CODE REVIEW

File: layers/core_checks/cc_device_memory.cpp [hunk 1]
```diff
@@ -92,6 +92,33 @@ bool CoreChecks::ValidateMemoryIsBoundToImage(const LogObjectList &objlist, cons
...
+    const vvl::Buffer &buffer_b = *accel_struct_b.buffer_state;
+
+    const sparse_container::range<VkDeviceSize> range_a(accel_struct_a.create_infoKHR.offset, accel_struct_a.create_infoKHR.size);
+    const sparse_container::range<VkDeviceSize> range_b(accel_struct_b.create_infoKHR.offset, accel_struct_b.create_infoKHR.size);
+
+    if (const auto [memory, overlap_range] = buffer_a.GetResourceMemoryOverlap(range_a, &buffer_b, range_b);
+        memory != VK_NULL_HANDLE) {
+        objlist.add(accel_struct_a.Handle(), buffer_a.Handle(), accel_struct_b.Handle(), buffer_b.Handle());
+
+        skip |= LogError(vuid, objlist, function_loc,
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_device_memory.cpp:111]: you can just go `loc_a.function` here instead and then you don't need to pass in `function_loc`

File: layers/error_message/logging.h [hunk 2]
```diff
@@ -56,6 +56,11 @@ struct LogObjectList {
         object_list.emplace_back(object, ConvertCoreObjectToVulkanObject(VkHandleInfo<HANDLE_T>::kVkObjectType));
     }
 
+    template <typename... HANDLE_T>
+    void add(HANDLE_T... objects) {
```
Comment (spencer-lunarg) [inline layers/error_message/logging.h:60]: honestly, I realize I should have done this a while ago! there are for sure many other functions I can now go back and do the same pattern! (thanks for adding this)

**Review by spencer-lunarg — APPROVED** [review]
With removal of extra `const Location& function_loc`

File: layers/core_checks/cc_device_memory.cpp (hunk 1 above)
Comment (arno-lunarg) [reply layers/core_checks/cc_device_memory.cpp:111]: Ok, will change this in a subsequent PR, I do not want to go through CI again x)

**Pull request merged**